atomic_actions.py

```diff
@@ -80,31 +82,33 @@ def attack_territory(from_territory, to_territory, troops_to_attack_with, verbos
             attacker_rolls, defender_rolls, bag_size = get_dice_bag(troops_to_attack_with, defender_troops, verbose)
 
         for k in range(2):
-            if i + k < len(attacker_rolls) and j + k < len(defender_rolls):
+            if i + k < len(attacker_rolls) and j + k < len(
+                    defender_rolls) and troops_to_attack_with > 0 and defender_troops > 0:

```

Comment:
`troops_to_attack_with > 0 and defender_troops > 0`. Without these, they may go under zero and it does not make sense. This bug was discovered by adding the test`test_failed_attack_with_three`.